stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -282,6 +283,14 @@ public fun <T : Any, R : Any> Result<T>.flatMap(func: (T) -> Result<R>): Result<
     }
 }
 
+public fun <T : Any, R : Any> Result<T>.flapMapCall(func: (T) -> Call<R>): Result<R> {
+    return if (isSuccess) {
+        func(data()).execute()
+    } else {
+        Result.error(error())
+    }
+}
```

Comment:
```suggestion
public fun <T : Any, R : Any> Result<T>.flapMapCall(func: (T) -> Call<R>, callback: Call.Callback<R>) {
    if (isSuccess) {
        func(data()).enqueue {
            callback.onResult(it)
        }
    } else {
        callback.onResult(Result.error(error()))
    }
}
```

Reply:
We also need somehow to cancel this child `Call<R>` if parent `Call<T>` gets cancelled